core/slice.go

```diff
@@ -586,18 +578,31 @@ func (sl *Slice) updatePhCacheFromDom(pendingHeader types.PendingHeader, termini
 }
 
 // updatePhCache updates cache given a pendingHeaderWithTermini with the terminus used as the key.
-func (sl *Slice) updatePhCache(pendingHeaderWithTermini types.PendingHeader, inSlice bool) {
+func (sl *Slice) updatePhCache(pendingHeaderWithTermini types.PendingHeader, inSlice bool, localHeader *types.Header) {
+	sl.phCacheMu.Lock()
+	defer sl.phCacheMu.Unlock()
+
+	var exists bool
+	if localHeader != nil {
+		termini := sl.hc.GetTerminiByHash(localHeader.ParentHash())
+		pendingHeaderWithTermini, exists = sl.readPhCache(termini[c_terminusIndex])
+		if exists {
+			pendingHeaderWithTermini.Header = sl.combinePendingHeader(localHeader, pendingHeaderWithTermini.Header, common.ZONE_CTX, true)
+		}
+	}
+
 	// Update the pendingHeader Cache
 	oldPh, exist := sl.readPhCache(pendingHeaderWithTermini.Termini[c_terminusIndex])
 	var deepCopyPendingHeaderWithTermini types.PendingHeader
 	newPhEntropy := pendingHeaderWithTermini.Header.CalcPhS()
 	deepCopyPendingHeaderWithTermini = types.PendingHeader{Header: types.CopyHeader(pendingHeaderWithTermini.Header), Termini: pendingHeaderWithTermini.Termini}
 	deepCopyPendingHeaderWithTermini.Header.SetLocation(common.NodeLocation)
 	deepCopyPendingHeaderWithTermini.Header.SetTime(uint64(time.Now().Unix()))
+
 	if exist {
 		// If we are inslice we will only update the cache if the entropy is better
```

Comment:
Do we want to update this comment as well?
Also can you provide brief explanation as to why its necessary now to update if entropy is equal?

Reply:
Answered in person

Reply:
Added comment to explain it too